### ISSUE TITLE
LayoutKit cells not updating in TableView and CollectionView examples

### ISSUE THREAD

**Issue by domasn** [linkedin/LayoutKit#133]
`4.0.2` release with no modifications.

Exampe of [`TableViewControllerFeedItemLayoutKitView`](https://github.com/linkedin/LayoutKit/blob/master/LayoutKitSampleApp/Benchmarks/TableViewController.swift#L18) uses [`FeedItemLayoutKitView`](https://github.com/linkedin/LayoutKit/blob/master/LayoutKitSampleApp/Benchmarks/FeedItemLayoutKitView.swift#L13) as a view wrapped inside a [`TableCell`](https://github.com/linkedin/LayoutKit/blob/master/LayoutKitSampleApp/Benchmarks/TableViewController.swift#L60).

On the first glance (on left) it seems to be working just fine - item with poster number `0` followed by item with poster number `1`. However after we scroll a bit item poster numbers lose their order (on right).. We see `5` followed by `0`:

![reuse bug](https://cloud.githubusercontent.com/assets/7569279/25217956/9f6f1814-25b1-11e7-9bff-4773e2bbaf8e.png)

-----

### Problem 1 - Reusing
After a bit of debugging we can see that [`layoutSubviews()` inside `FeedItemLayoutKitView`](https://github.com/linkedin/LayoutKit/blob/master/LayoutKitSampleApp/Benchmarks/FeedItemLayoutKitView.swift#L37) does not get called for reused cells leaving cells with outdated content.

This can be solved by adding  `setNeedsLayout()` at the end of [`setData(_:)` inside `FeedItemLayoutKitView`](https://github.com/linkedin/LayoutKit/blob/master/LayoutKitSampleApp/Benchmarks/FeedItemLayoutKitView.swift#L27).

### Problem 2 - Height
Even with `setNeedsLayout()` inserted, table view does not call [`intrinsicContentSize` on `FeedItemLayoutKitView`](https://github.com/linkedin/LayoutKit/blob/master/LayoutKitSampleApp/Benchmarks/FeedItemLayoutKitView.swift#L33) for every cell before display. Even though example uses `UITableViewAutomaticDimension` as row height, differnet height cells created with LayoutKit would simply not work.

-----

Please see simple [playground code with live examples in my gist](https://gist.github.com/domasn/58a42251a81e26ffda447e267b363ed2).

Thank you for working on this amazing project! 👏

**Comment by nicksnyder** [comment]
Thanks for the great bug report!

I don't think I will have time to take a look at this in any reasonable timeframe so PRs from anyone would be welcome.

@jingwei-huang1 @staguer

**Comment by jingwei-huang1** [comment]
From my first glance, I found it is because of autolayout. I will investigate this more.

**Comment by domasn** [comment]
@jingwei-huang1 Indeed it's a problem with `Autolayout`. I tried playing with it and managed to build a working solution. See [recent changes](https://gist.github.com/domasn/58a42251a81e26ffda447e267b363ed2/revisions#diff-08025799cfc5d9c1a252086722c6fcfe) in [the same gist file](https://gist.github.com/domasn/58a42251a81e26ffda447e267b363ed2).

In summary - I added a height constraint to `wrappedContentView` that's constant needs to be updated to the height of layed-out view.
